larsim/EventGenerator/POTaccumulator_module.cc

```diff
@@ -0,0 +1,307 @@
+  std::string fSummaryOutputCategory; ///< Name of the main stream for output.
+  std::string fRunOutputCategory; ///< Name of the run stream for output.
+
+  // -- END -- Configuration variables -----------------------------------------
+
+
+  // -- BEGIN -- Internal cache variables --------------------------------------
+
+  /// Count of subrun fragments with POT information.
+  std::map<art::SubRunID, unsigned int> fPresentSubrunFragments;
+
+  /// Count of subrun fragments without POT information.
+  std::map<art::SubRunID, unsigned int> fMissingSubrunFragments;
+
+  /// Partial count of POT in the run, per run.
+  std::map<art::RunID, sumdata::POTSummary> fRunPOT;
+
+  // -- END -- Internal cache variables ----------------------------------------
+
+
+  /// Prints the list of subruns with partial or missing POT information.
+  void printMissingSubrunList() const;
+
+  /// Prints the list of POT per run.
+  void printRunSummary() const;
+
+  /// Prints the total POT summary.
+  void printSummary() const;
+
+
+  /// Converts the information from `POT` in a compact string.
+  static std::string to_string(sumdata::POTSummary const& POT);
+
+}; // class sim::POTaccumulator
+
+
+//------------------------------------------------------------------------------
+//---  module implementation
+//---
+//------------------------------------------------------------------------------
+sim::POTaccumulator::POTaccumulator(Parameters const& config)
+  : EDAnalyzer(config)
+  , fPOTtag(config().SummaryTag())
+  , fSummaryOutputCategory(config().SummaryCategory())
+  , fRunOutputCategory(config().RunSummaryCategory())
+  {}
+
+
+//------------------------------------------------------------------------------
+void sim::POTaccumulator::beginSubRun(art::SubRun const& subRun) {
+
+  auto const& ID = subRun.id();
+
+  //
+  // get the information from the subrun and update the subrun counts
+  //
+  art::Handle<sumdata::POTSummary> summaryHandle;
+  if (!subRun.getByLabel(fPOTtag, summaryHandle)) {
+    ++fMissingSubrunFragments[ID];
+    mf::LogDebug(fSummaryOutputCategory) << "Fragment of subrun " << ID
+      << " has no '" << fPOTtag.encode() << "' POT summary.";
+    return;
+  }
+
+  ++fPresentSubrunFragments[ID];
+
+  //
+  // accumulate the information by run
+  //
+  auto const& runID = subRun.getRun().id();
+
+  sumdata::POTSummary const& subRunPOT = *summaryHandle;
+
+  fRunPOT[runID].aggregate(subRunPOT);
+  MF_LOG_TRACE(fSummaryOutputCategory) << "Fragment #"
+    << fPresentSubrunFragments[ID] << " of subrun " << ID
+    << ": " << sim::POTaccumulator::to_string(subRunPOT)
+    ;
+
+} // sim::POTaccumulator::analyze()
+
+
+//------------------------------------------------------------------------------
+void sim::POTaccumulator::endJob() {
+
+  //
+  // print the run summary
+  //
+
+  if (!fRunOutputCategory.empty()) {
+
+    if (!fMissingSubrunFragments.empty()) printMissingSubrunList();
+
+    printRunSummary();
+
+  } // if run report requested
```

Comment:
The comment on this line seems unnecessary.

Reply:
Reduced. I use `// if` as a graphical reminder of the block.